### DISCUSSION TITLE
Pause song after break if tablet pen is not in proximity

### DISCUSSION THREAD

**Discussion by slopslug** [ppy/osu#20350] [Ideas] [answered]
Title, this is something that I'm really missing from stable and would love to see it in lazer!

OpenTabletDriver already supports this (Tablets > Tablet Debugger > NearProximity or see the gif below) so I don't think it'd be that hard to implement.

![Tablet_Debugger-1663361611](https://user-images.githubusercontent.com/47635446/190730932-b345ac2a-bb3c-4083-96ba-a5467cfd91a5.gif)

**Comment by peppy** [comment]
I'm confused. I don't remember implementing this in stable.

**Reply by slopslug** [reply]
Sorry for the confusion! I meant this as something that I wish was in stable.

**Comment by ItsShamed** [comment, marked as answer]
Might not be a good idea imo, especially for people who hover or tap with their pen. The game might pause by accident each time you lift your pen a little bit too much. 
Also the `NearProximity` property seems to be a thing only for Wacom Intuos tablets (I don't have this on my XP-Pen tablet):
![image](https://user-images.githubusercontent.com/39100084/190866479-5525433d-bc4c-43fa-8589-9fde6fc8ae5c.png)

One might implement this by listening for `OutOfRangeReport` through `TabletDriver`'s `DeviceReported` event.
But again, I'm not too fond of this idea.

**Reply by slopslug** [reply]
I hover and have been for some time now. However, I don't hover on the brink of my NearProximity range and I'd like to assume that most people don't either. The choice is something to consider which is why I would add this with a toggle.

Also, this should not happen if a break is still ongoing, but rather if the break has ended or during a longer period between notes which are still technically breaks but not specified as such.

**Reply by ItsShamed** [reply]
> I'd like to assume that most people don't either.

Very dangerous assumption then. You should know that tablets do not have the same hover distance: while a CTL-480 has a hover distance of 17mm, a CTL-490 only has 11mm. This might sound like it's not much of a difference, but for some people those few millimeters can be crucial.

> Also, this should not happen if a break is still ongoing, but rather if the break has ended or during a longer period between notes which are still technically breaks but not specified as such.

Then what's even the point of breaks? With breaks you have plenty of time to pause or reset your grip or do whatever you want. 
And for really "time-spaced" notes you can just pause manually between them it's the same thing, I really don't see the point of having to lift your pen for this.

I think we should wait for more opinions, I don't know how to feel about this.

**Reply by smoogipoo** [reply]
Yeah. This is not happening. I personally hover very close to `NearProximity` and when changing grip/coming back after a long time I tend to go beyond it and have my aim "skip".

From another perspective, breaks need to be a deliberate action at all times with no room for failure/false positives.
